File: backend/app/api/routes/categories.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.api.deps import get_db
from app.schemas.category import CategoryCreate, CategoryResponse, CategoryUpdate
from app.services import category_service

router = APIRouter()


@router.get("/categories", response_model=list[CategoryResponse])
def list_categories(db: Session = Depends(get_db)) -> list[CategoryResponse]:
    return category_service.get_categories(db)


@router.post("/categories", response_model=CategoryResponse, status_code=201)
def create_category(
    data: CategoryCreate, db: Session = Depends(get_db)
) -> CategoryResponse:
    return category_service.create_category(db, data)


@router.patch("/categories/{category_id}", response_model=CategoryResponse)
def update_category(
    category_id: int, data: CategoryUpdate, db: Session = Depends(get_db)
) -> CategoryResponse:
    return category_service.update_category(db, category_id, data)

File: frontend/src/api/categories.ts
import { apiClient } from './client'
import type { Category, CategoryCreate, CategoryUpdate } from '../types/category'

export const getCategories = (): Promise<Category[]> => apiClient.get('/categories')
export const createCategory = (data: CategoryCreate): Promise<Category> => apiClient.post('/categories', data)
export const updateCategory = (categoryId: number, data: CategoryUpdate): Promise<Category> =>
	apiClient.patch(`/categories/${categoryId}`, data)
